### DISCUSSION TITLE
`[data-testid]` as CSS selectors for customization

### DISCUSSION THREAD

**Discussion by leomp12** [StaticJsCMS/static-cms#852] [Q&A] [answered]
_data-testid_ attribute is everywhere on https://demo.staticcms.org , for example `<img src="..." data-testid="image">` on image widget thumbnail.
Does anyone know if this attribute is only set in the demo CMS? I was thinking about using it as selectors for some CSS overwrites.

**Comment by KaneFreeman** [comment, marked as answer]
The `data-testid` tag is used for internal automated testing purposes. While they can be used, I would not recommend it. They made change at any time without warning.

**Reply by leomp12** [reply]
Any chances on creating more stable element selectors thought? Data attributes like `data-widget="image"` or class names (even using tailwind, just for external customization)...

**Comment by leomp12** [comment]
https://github.com/StaticJsCMS/static-cms/issues/867#issuecomment-1719560491 👍🏾
